lib/reek/smells/smell_detector.rb

```diff
@@ -38,10 +39,13 @@ def contexts
       end
 
       def run_for(context)
-        return [] unless enabled_for?(context)
         return [] if exception?(context)
 
-        sniff(context)
+        if enabled_for?(context)
+          sniff(context)
+        else
+          ::Reek::Smells::UnneededSuppression.new(context, self).sniff
```

Comment:
Actually, in view of my comment below, I think the end goal should be something like:
```ruby
if enabled_for?(context)
  sniff(context)
else
  sniff_unneeded_suppression(context)
end

# [...]

def sniff_unneeded_suppression(context)
  return [] if sniff(context)
  # return correct unneeded suppression smell warnings here
end
```

The 'return correct unneeded suppression smell warnings here' will need some work, since `smell_warning` is currently a private method.